src/parsehdr.c

```diff
@@ -358,15 +361,19 @@ cr_package_from_header(Header hdr,
                     }
 
                     // Skip package primary files
-                    if (g_hash_table_lookup_extended(filenames_hashtable, filename, NULL, NULL)) {
+                    if (*filename == '/' && g_hash_table_lookup_extended(filenames_hashtable, filename, NULL, NULL)) {
                         if (cr_is_primary(filename)) {
                             continue;
                         }
                     }
 
                     // Skip files which are provided
-                    if (g_hash_table_lookup_extended(provided_hashtable, filename, NULL, NULL)) {
-                        continue;
+                    gpointer pvalue;
+                    if (g_hash_table_lookup_extended(provided_hashtable, filename, NULL, &pvalue)) {
+                        struct ap_value_struct *ap_value = pvalue;
+                        if (!ap_value->flags || !ap_value->flags[0] || !flags || !flags[0] ||
+                            (!g_strcmp0(ap_value->flags, flags) && !g_strcmp0(ap_value->version, full_version)))
+                           continue;
```

Comment:
As the condition is on two lines and indentation could be confusing, could you please put curly brackets around the ``continue;`` single line scope?